gateway/Caddyfile

```diff
@@ -0,0 +1,51 @@
+http://:3000 {
+	log {
+		level INFO
+	}
+
+	# CORS
+
+	@OPTIONS {
+		method OPTIONS
+	}
+	header {
+		Access-Control-Allow-Origin *
+		Access-Control-Allow-Credentials true
+		Access-Control-Allow-Methods *
+		Access-Control-Allow-Headers *
+		defer
+	}
+	handle @OPTIONS {
+		respond 204
+	}
+
+	# Unprotected routes
+
+	handle /users {
+		reverse_proxy http://agartex-authentication:3100
+	}
+
+	handle /sessions {
+		reverse_proxy http://agartex-authentication:3100
+	}
+
+	# Protected routes
+
+	handle * {
+		forward_auth http://agartex-authentication:3100 {
+			uri /users
+			copy_headers X-User-Id
+		}
+
+		handle /compile {
+			reverse_proxy http://agartex-compilation:3200
+		}
```

Comment:
Shouldn't be protected

Reply:
Also port should be 3300

Reply:
It's for demo purpose, I keep it protected